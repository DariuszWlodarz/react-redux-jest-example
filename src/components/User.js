import React from 'react';
import PropTypes from 'prop-types';


const User = ({user, onClickHandler}) => (
  <div onClick={() => onClickHandler(user)}>{`${user.id} ${user.email}`}</div>
);

User.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    email: PropTypes.string.isRequired
  }).isRequired,
  onClickHandler: PropTypes.func.isRequired
};

export default User;