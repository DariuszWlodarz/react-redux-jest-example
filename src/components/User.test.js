import React from 'react';
import User from './User';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

const user = {
  id: 1,
  email: 'user1@example.com'
};

test('snapshot', () => {
  const component = shallow(
    <User user={user} onClickHandler={() => {}}/>
  );

  expect(toJson(component)).toMatchSnapshot();
});

test('click', () => {
  const onClick = jest.fn();
  const component = shallow(<User user={user} onClickHandler={onClick} />);
  component.simulate('click');

  expect(onClick).toHaveBeenCalledTimes(1);
  expect(onClick).toHaveBeenCalledWith(user);
});

