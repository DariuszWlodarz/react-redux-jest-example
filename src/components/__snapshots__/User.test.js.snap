// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  onClick={[Function]}
>
  1 user1@example.com
</div>
`;
